hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -680,7 +680,9 @@ def should_generate_more():
 
         count = 0
         while should_generate_more() and (
-            count < 10 or self.health_check_state is not None
+            count < 10
+            or self.health_check_state is not None
+            or len(self.target_selector) == 0
```

Comment:
In keeping with the general habit of erring on the side of overcommenting in conjecture, it'd be nice to have a one-line sentence saying why we need the `target_selector` check. (I know this bit of the code is in general currently undercommented, but continuous improvement!)